Docs: Issue #17: Update README with build changes

nattvara opened the pull request:
This PR improves the general structure of the README, fixes outdated information about how to run app with intellij, and adds some new information about how to run the main program

nattvara committed:
add TOC and fix markdown lint issues in readme

nattvara committed:
restructure the sections of the readme

nattvara committed:
add section about development workflow

nattvara committed:
add formatting and update instructions

nattvara committed:
add run, build and test instructions

siberov commented:
`.DS_store` files under `docs` and `docs/img` should be removed.

siberov commented on README.md:
```diff
@@ -1,25 +1,154 @@
+
+Welcome to the first assignment in the course Software Engineering Fundamentals (DD2480). We will get you up to speed with what this assignment is about and what we have done.
+
+Furthermore, how to run and debug the program on your own computer and run the test suite.
```
Minor grammar mistake. Missing subject. Maybe "We will also show you how to [...]".

siberov committed:
Fix: Minor README changes and file cleanup (#66)

* Minor README changes and file cleanup

The intro to the README was changed somewhat, and some spelling and
grammar fixes were done throughout.

Unnecessary .DS_Store files were purged.

* Make changes suggested by SamuSoft
